lib/style-engine/class-wp-style-engine-gutenberg.php

```diff
@@ -0,0 +1,166 @@
+<?php
+/**
+ * WP_Style_Engine class
+ *
+ * Generates classnames and block styles.
+ *
+ * @package Gutenberg
+ */
+
+/**
+ * Singleton class representing the style engine.
+ *
+ * Consolidates rendering block styles to reduce duplication and streamline
+ * CSS styles generation.
+ *
+ * @since 6.0.0
+ */
+class WP_Style_Engine_Gutenberg {
+	/**
+	 * Container for the main instance of the class.
+	 *
+	 * @since 5.5.0
```

Comment:
I was going to say, should this be `@since 6.0.0`, but because the entire class is new, I imagine we don't need to add `@since`? (I wasn't too sure what the practice is there in core 🤔)

Reply:
Copy paste job 😄